### ISSUE TITLE
PermissionError Errno 13 on RoBERTa disambiguate_pronoun()

### ISSUE THREAD

**Issue by de1019** [pytorch/fairseq#2067]
#### What is your question?

My ultimate goal is to use RoBERTa to disambiguate my own sentences. I installed PyTorch, fairseq (without SacreBleu to avoid Windows error), and spaCy yesterday. But when I use RoBERTa's disambiguate_pronoun() I get the PermissionError [Errno 13] on a temp file as shown below.  What am I doing wrong? 

#### Code

>>> roberta = torch.hubroberta = torch.hub.load('pytorch/fairseq', 'roberta.large.wsc', user_dir='examples/roberta/wsc')
Using cache found in C:\Users\Daniel/.cache\torch\hub\pytorch_fairseq_master
| dictionary: 50265 types
>>>
>>> roberta.disambiguate_pronoun('The city councilmen refused the demonstrators a permit because [they] feared violence.')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\Daniel/.cache\torch\hub\pytorch_fairseq_master\fairseq\models\roberta\hub_interface.py", line 204, in disambiguate_pronoun
    return self.task.disambiguate_pronoun(self.model, sentence, use_cuda=self.device.type == 'cuda')
  File "C:\Users\Daniel/.cache\torch\hub\pytorch_fairseq_master\fairseq\..\examples/roberta\wsc\wsc_task.py", line 223, in disambiguate_pronoun
    dataset = self.build_dataset_for_inference(sample_json)
  File "C:\Users\Daniel/.cache\torch\hub\pytorch_fairseq_master\fairseq\..\examples/roberta\wsc\wsc_task.py", line 217, in build_dataset_for_inference
    return_only=True,
  File "C:\Users\Daniel/.cache\torch\hub\pytorch_fairseq_master\fairseq\..\examples/roberta\wsc\wsc_task.py", line 123, in load_dataset
    for sentence, pronoun_span, query, label in wsc_utils.jsonl_iterator(data_path):
  File "C:\Users\Daniel/.cache\torch\hub\pytorch_fairseq_master\fairseq\..\examples/roberta\wsc\wsc_utils.py", line 94, in jsonl_iterator
    with open(input_fname) as fin:
PermissionError: [Errno 13] Permission denied: 'C:\\Users\\Daniel\\AppData\\Local\\Temp\\tmpuw5rwuav' 
>>>

#### What have you tried?

- Attempted running Anaconda3 prompt as Adminstrator
- Uninstall/Reinstall spaCy to 2.2.4
- Tried turning off Read Only on the Temp folder, but the file is created and deleted during each run.
- Tried a variety of sentences, with or without marking the antecedent. All legal entries are getting the same error.

#### What's your environment?

 - fairseq Version (0.9.0):
 - PyTorch Version (e.g., 1.5.0)
 - OS (Windows 10 Home 64-bit):
 - How you installed fairseq (downloaded zip, removed SacreBleu reference in setup.py, `pip install C:\Users\Daniel\.cache\torch\hub\fairseq_master'):
 - Build command you used (if compiling from source):
 - Python version: 3.7.6 via Anaconda CMD.exe Prompt 0.1.1
 - CUDA/cuDNN version: None
 - GPU models and configuration: 2x NVIDEA GeForce GTX 970 in SLI
 - Any other relevant information:
 ---- Total physical memory 32GB - DDR4.
 ---- Available physical memory 25GB.

**Comment by myleott** [comment]
For that code, we write the input sentence into a temporary file and read it back (it’s kind of hacky). It seems there’s some issue accessing the temporary file after it’s been written though.

Here is the relevant piece of code: https://github.com/pytorch/fairseq/blob/master/examples/roberta/wsc/wsc_task.py#L212

Can you try debugging your environment a bit to figure out why you can’t read back the file?

You can even try something self-contained and independent of fairseq:
```python
with tempfile.NamedTemporaryFile(buffering=0) as h:
    h.write(“hello world\n”.encode(“utf-8”))
    with open(h.name) as h_in:
        for line in h_in:
            print(line)
```

**Comment by ahsmourad** [comment]
Hi myleott,

I have ran into the same issue on Windows 10. After debugging the code sample you shared, it generated the same error. Then I went to check the [tempfile.NamedTemporaryFile](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile) documentation. It says: "Whether the name can be used to open the file a second time, while the named temporary file is still open, varies across platforms (it can be so used on Unix; it cannot on Windows NT or later)." It seems to be a common issue on Windows that Python hasn't fixed for years, check [this](https://bugs.python.org/issue22107) out. Is there a way to fix that?
